src/Assets/Attributes.php

```diff
@@ -29,19 +31,29 @@ public function asset(Asset $asset)
     }
 
     /**
-     * Get the dimensions of an asset.
+     * Get the attributes of an asset.
      *
      * @return array
      */
     public function get()
     {
+        if ($this->asset->isAudio()) {
+            return $this->getAudioAttributes();
+        }
+
         if ($this->asset->isImage()) {
-            return $this->getImageDimensions();
-        } elseif ($this->asset->isSvg()) {
-            return $this->getSvgDimensions();
+            return $this->getImageAttributes();
+        }
+
+        if ($this->asset->isSvg()) {
+            return $this->getSvgAttributes();
+        }
+
+        if ($this->asset->isVideo()) {
+            return $this->getVideoAttributes();
         }
 
-        return [null, null];
+        return [null, null, null];
```

Comment:
When this was just dealing with dimensions, the `[width, height]` made sense.

Now that it's returning potentially a whole bunch of stuff (I could see more than length in the future being added), I don't think this works.

Can we change this to an array?

```php
return [
 'width' => $width,
 'height' => $height,
 'length' => $length
 'etc' => $etc,
];
```

(or `compact('width', 'height', 'length', 'etc')`)

I know this one comment affects a lot of other lines in this PR. 😬

Reply:
Ya I was thinking the same thing

Reply:
this has been done